internal/adapters/secondary/identity/identity.go

```diff
@@ -0,0 +1,164 @@
+package identity
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Businge931/sba-user-accounts/internal/core/domain"
+	"github.com/Businge931/sba-user-accounts/internal/core/errors"
+	"github.com/Businge931/sba-user-accounts/internal/core/ports"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type identityProvider struct {
+	authRepo ports.AuthRepository
+	userRepo ports.UserRepository
+	tokenSvc ports.TokenService
+	logger   ports.Logger
+}
+
+func NewIdentityProvider(
+	authRepo ports.AuthRepository,
+	userRepo ports.UserRepository,
+	tokenSvc ports.TokenService,
+	logger ports.Logger,
+) ports.IdentityService {
+	return &identityProvider{
+		authRepo: authRepo,
+		userRepo: userRepo,
+		tokenSvc: tokenSvc,
+		logger:   logger,
+	}
+}
+
+func (svc *identityProvider) RegisterSvc(req domain.RegisterRequest) (*domain.User, error) {
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(req.Password), bcrypt.DefaultCost)
+	if err != nil {
+		return nil, fmt.Errorf("failed to hash password: %w", err)
+	}
+
+	// Create user with UUID
+	user := domain.NewUser(req.Email, req.FirstName, req.LastName)
+	user.HashedPassword = string(hashedPassword)
+	user.ID = fmt.Sprintf("%d-%d", time.Now().UnixNano(), time.Now().Nanosecond())
+
+	if svc.authRepo != nil {
```

Comment:
No need for this check here. Such a check, if you need it, should only be done during service start up